smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/CodegenVisitor.kt

```diff
@@ -47,10 +47,18 @@ class CodegenVisitor(context: PluginContext) : ShapeVisitor.Default<Unit>() {
     init {
         val classLoader = context.pluginClassLoader.orElse(javaClass.classLoader)
         LOGGER.info("Discovering KotlinIntegration providers...")
-        integrations = ServiceLoader.load(KotlinIntegration::class.java, classLoader)
-            .also { integration -> LOGGER.info("Loaded KotlinIntegration: ${integration.javaClass.name}") }
+
+        val integrationsBeforeReplacement = ServiceLoader.load(KotlinIntegration::class.java, classLoader)
+            .onEach { integration -> LOGGER.info("Loaded KotlinIntegration: ${integration.javaClass.name}") }
             .filter { integration -> integration.enabledForService(context.model, settings) }
-            .also { integration -> LOGGER.info("Enabled KotlinIntegration: ${integration.javaClass.name}") }
+            .onEach { integration -> LOGGER.info("Enabled KotlinIntegration: ${integration.javaClass.name}") }
```

Comment:
correctness

I think this log message is misleading as any particular integration may be removed in the subsequent code.  Maybe "loaded" and then "enabled" in the final stanza?  Generally I think it's noisy if we are saying anything about an unused integration except that it's unused in logs.